src/components/ContentWidgets/ToDoList/TodoListEditor.tsx

```diff
@@ -1,51 +1,61 @@
 import React from 'react';
+// use shortid for generating unique keys
+import shortid from 'shortid';
 import { WidgetEditorProps } from '../../ContentMapping/ContentMapping';
 import { TodoItem, TodoList } from './TodoList';
 import "./TodoStyle.css";
 
-// an input box for adding a new todo item or checking/unchecking
-// existing todo items.
+//
 //
 // Last Modified
 // Max Zhou
 // July 6, 2019
 //
+
+/**
+ * an input box for adding new todo items or marking existing todos as complete.
+ *
+ * Last Modified
+ * Max Zhou
+ * July 6, 2019
+ * */
 export const TodoListEditor: React.FC<WidgetEditorProps> = ({
   editedContent,
   setEditedContentOnChange
 }) => {
-	// get the list
 	let currentList: TodoItem[] = editedContent.todoList_items || [];
 
-	let list_html = currentList.map((item: TodoItem, index: number) => <li className={item.check} key={index}>{item.name}<button type="button" onClick={(e) => handleCheck(index)}>done</button></li>);
+	// make a local copy with spread
+	let localList = [...currentList];
```

Comment:
you would probably want this to be a state. 

`const [localList, setLocalList] = useState<TodoItem[]>(editedContent.todoList_items ? ([...(editedContent.todoList_items)] as TodoItem[]): [])`